### ISSUE TITLE
Map Stream MJPEG instead of series of images

### ISSUE THREAD

**Issue by tonysprenk** [Tasshack/dreame-vacuum#724]
**Is your feature request related to a problem? Please describe.**
I want to add the map stream to HomeKit. This works but only for the preview image. I believe this is because the stream created by this integration is not an actual MJPEG stream but rather just a series of images. 

**Describe the solution you'd like**
It would be nice if the stream of the map image was in H264 (HomeKit compatible) or at least actual MJPEG rather than a series of images so that home assistant can decode it properly for HomeKit. 

**Describe alternatives you've considered**
I've tried encoding with go2ryc but it does not work as the stream provided by this integration is not actual MJPEG. 

**Additional context**
Thank you for the great integration!

**Comment by tonysprenk** [comment]
Yes I'm using:

``` 
{HOMEASSISTANT}/api/camera_stream_proxy/camera.{ENTITY}_map?token={TOKEN}
``` 

But I'm actually injecting a long lived access token authorization header because the token in the url refreshes every so often. Either way the stream I believe is just a series of images.

**Comment by tonysprenk** [comment]
The result is the same. When I put this stream into HomeKit I get an error in home assistant saying:

``` 
Camera has no stream source
``` 
When I put the camera component from the Dreame integration I get the same error.

**Comment by tonysprenk** [comment]
Yes I saw this as well but when I try to convert with go2rtc I still get no camera stream found.

**Comment by tonysprenk** [comment]
Whatever I do: convert with ffmpeg, convert with go2rtc, use the built in conversion in the HomeKit integration. I always get a "source has no stream error". Not sure what my next steps should be. Any suggestions would be highly appreciated. I think there would be quite a bit of interest from people using HomeKit. Thanks!

**Comment by tonysprenk** [comment]
I think the fact that they are png images is the clue. When I do the following command in terminal outside of home assistant it works as expected. 

```
ffmpeg -re -loop 1 -headers "Authorization: Bearer long_lived_access_token" -i "http://homeassistant.local:8123/api/camera_proxy/camera.x40_ultra_complete_map" -vf "fps=1" -vcodec libx264 -r 15 -pix_fmt yuv420p output.mp4
``` 


Note the loop argument. FFMPEG needs to be told to keep looping even if the image is not refreshed constantly. Does the Dreame integration only refresh the image when a change is detected? Or does it serve a new image every second for example? The only thing I need to do now is to find a way to implement the above command in home assistant. I haven't been successful yet as I'm getting 400 errors.

**Comment by incline02** [comment]
Hi @Tasshack I'm following up from your reply [here](https://community.home-assistant.io/t/valetudo-map-card-installation-instructions/247329/31?u=solarsol).

Like @tonysprenk, I've tried it all without success.
The url I'm trying to access is:
`http://192.168.xxx.xxx:8123/api/camera_proxy/camera.bot_map?token={{ state_attr('camera.bot_map', 'access_token') }}`

Giving me this for FFMPEG
camera:
  - platform: ffmpeg
    name: ffmpeg live view
    input: http://192.168.xxx.xxx:8123/api/camera_proxy/camera.bot_map?token={{ state_attr('camera.bot_map', 'access_token') }}

When adding ffmpeg camera to the dashboard the picture is blank: 
<img width="787" alt="image" src="https://github.com/user-attachments/assets/1a76a861-acc8-4245-8bce-fec0ed54d783">

For Camera Proxy:
it works, I get the feed but in homekit, it still won't work live. only the preview

For go2rtc
It doesn't take dynamic url so I replace "token=abc" with "token=temporary_one_found_in_states" and I get: "Unsupported header: 89504e47

Using Generic Camera
I've suceeded in creating a Generic Camera feed using the dynamic url above but exactly the same problems occur.

Lastly, I did also create a life long token giving me the url:
http://192.168.xxx.xxx:8123/api/camera_proxy_stream/camera.bot_map#header=Authorization: Bearer life_long_token

but I don't even know where to test it since it has spaces in it, nor the syntax it should have in the config file, the platform to use it in, etc etc

I hope that brings you more info. Any help?
